### DISCUSSION TITLE
Using fetch in Javascript to a CORS enabled website

### DISCUSSION THREAD

**Discussion by edguy99** [community/community#111246] [Programming Help]
### Body

Working on a Neutrino tracking project that contains billions of events (to much to store on github).

https://github.com/edguy99/icecube

I want to fetch data from an outside source, but get back only this with no data? Appears the CORS is ok, but why cant I get some data in there?

{type: 'cors', url: 'https://crumbsoftware.com:3028/eventwanted-647743931-false', redirected: false, status: 200, ok: true, …}

Anyone able to help on a fetch in javascript?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by amankr1469** [comment]
First, check the Network tab to inspect CORS headers. If nothing helps then try using Axios and check if you are getting the data from API call or not.

**Comment by mrspecht** [comment]
Here is an example of an async function to fetch data from an API. The options object contains info about CORS

```
const url = 'https://api.andrespecht.dev/movies';

const options = {
  method: 'GET',
  mode: 'cors'
};

async function getMovies() {
  try {
    const response = await fetch(url, options);
    if (!response.ok) {
      throw new Error(`${response.statusText} (${response.status})`);
    }
    // Parsing the reponse as JSON
    const data = await response.json();
    // Printing the movies
    console.log(data.response);
  } catch(error) {
    console.log(error);
  }
};

getMovies();
```
